fineract-core/src/main/java/org/apache/fineract/infrastructure/core/service/database/JavaType.java

```diff
@@ -91,9 +91,7 @@ public enum JavaType {
     OBJECT(Object.class), //
     ;
 
-    public static final JavaType[] VALUES = values();
-
-    private static final Map<Class<?>, JavaType> BY_TYPE = Arrays.stream(VALUES).filter(e -> e.type != null)
+    private static final Map<Class<?>, JavaType> BY_TYPE = Arrays.stream(values()).filter(e -> e.type != null)
```

Comment:
Actually it does make sense to have a static VALUES array, because values() is dynamic. If it only used here for the static mapper then you can remove, but if it is used anywhere else then it worths to keep.

Reply:
See my earlier comment. Again, kind of defeats the purpose of an enum. With an enum you have type safety built-in, but then we bolt all those integers on top of it and create a parallel structure that we could have simpler with constants, why bother with enums?!?

Reply:
Reading this again... I might have commented a bit off topic here. Was still in the "enum" thinking that was discussed earlier. This here is related to your database mapping mechanics. Don`'t really have an opinion here other than: we have technically 2 other database mapping mechanics in place (JPA, Spring Data JDBC) that kind of - at least partially - do a lot of what you do here (type mapping). It feels a bit redundant, but I have to admit that I didn't follow exactly what requirements you had on your list and/or if there is maybe no other way to it.
Again, not relevant for the release.